subsys/logging/backends/log_backend_net.c

```diff
@@ -233,6 +242,28 @@ bool log_backend_net_set_addr(const char *addr)
 	return ret;
 }
 
+bool log_backend_net_set_ip(const struct sockaddr *addr)
+{
+	bool ret = check_net_init_done();
+
+	if (!ret) {
+		return ret;
+	}
+
+	memcpy(&server_addr, addr, sizeof(server_addr));
+
+	if (addr->sa_family == AF_INET) {
```

Comment:
I think we should overwrite the port number provided in `addr` if it's unspecified (`0`).